resharper/resharper-unity/src/Rider/UnityEditorFindRequestCreator.cs

```diff
@@ -0,0 +1,132 @@
+        {
+            var finder = mySolution.GetPsiServices().Finder;
+            var references = finder.FindAllReferences(declaredElement).OfType<IUnityYamlReference>();
+
+            var result = new List<FindUsageResult>();
+
+            foreach (var reference in references)
+            {
+                var request = CreateRequest(reference, selectedReference);
+                if (request != null)
+                    result.Add(request);
+            }
+
+            if (result.Count == 0)
+                return false;
+
+            if (focusUnity)
+            {
+                UnityFocusUtil.FocusUnity(myUnityHost.GetValue(t => t.UnityProcessId.Value));
+            }
+            
+            if (selectedReference != null)
+                myUnityHost.PerformModelAction(t => t.ShowGameObjectOnScene.Fire(CreateRequest(selectedReference, null)));
+            myUnityHost.PerformModelAction(t => t.FindUsageResults.Fire(result.ToArray()));
+            return true;
+        }
+        
+        private FindUsageResult CreateRequest([NotNull] IUnityYamlReference currentReference, [CanBeNull] IUnityYamlReference selectedReference)
+        {
+            var gameObjectDocument = currentReference.ComponentDocument.GetUnityObjectDocumentFromFileIDProperty(UnityYamlConstants.GameObjectProperty) ?? currentReference.ComponentDocument;
+
+            var sourceFile = gameObjectDocument?.GetSourceFile();
+            if (sourceFile == null)
+                return null;
+
+            var pathElements = UnityObjectPsiUtil.GetGameObjectPathFromComponent(currentReference.ComponentDocument).RemoveEnd("\\").Split("\\");
+            var rootIndices = new FrugalLocalList<int>();
+            
+            // Constructing path of child indices
+            UnityObjectPsiUtil.ProcessToRoot(gameObjectDocument,(currentGameObject, modification) =>
+                {
+                    int rootOrder = -1;
+                    var transform = UnityObjectPsiUtil.FindTransformComponentForGameObject(currentGameObject);
+                    if (modification != null)
+                    {
+                        if (!int.TryParse(UnityObjectPsiUtil.GetValueFromModifications(modification, transform.GetFileId(), UnityYamlConstants.RootOrderProperty)
+                            , out rootOrder))
+                            rootOrder = -1;
+                    }
+                    if (rootOrder == -1)
+                    {
+                        var rootOrderAsString = transform.GetUnityObjectPropertyValue(UnityYamlConstants.RootOrderProperty).AsString();
+                        if (!int.TryParse(rootOrderAsString, out rootOrder))
+                            rootOrder = -1;
+                    }
+                    rootIndices.Add(rootOrder);
+                });
+
+
+            if (!GetPathFromAssetFolder(sourceFile, out var pathFromAsset, out var fileName, out var extension))
```

Comment:
What about assets in packages? These might be in the `Packages` folder or in a readonly cache folder